tests/bluetooth/tester/src/btp_vcp.c

```diff
@@ -0,0 +1,580 @@
+		return;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_VOL_DOWN, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+static void mute(void)
+{
+	LOG_DBG("Mute");
+
+	if (bt_vcp_vol_rend_mute() != 0) {
+		tester_rsp(BTP_SERVICE_ID_VCS, VCS_MUTE, CONTROLLER_INDEX,
+			   BTP_STATUS_FAILED);
+		return;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_MUTE, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+static void unmute(void)
+{
+	LOG_DBG("Unmute");
+
+	if (bt_vcp_vol_rend_unmute() != 0) {
+		tester_rsp(BTP_SERVICE_ID_VCS, VCS_UNMUTE, CONTROLLER_INDEX,
+			   BTP_STATUS_FAILED);
+		return;
+	}
+
+	tester_rsp(BTP_SERVICE_ID_VCS, VCS_UNMUTE, CONTROLLER_INDEX,
+		   BTP_STATUS_SUCCESS);
+}
+
+static void vcs_state_cb(int err, uint8_t volume, uint8_t mute)
+{
+	if (err) {
+		LOG_DBG("VCP state cb err (%d)", err);
+	}
+	LOG_DBG("VCP state cb");
+}
+
+static void vcs_flags_cb(int err, uint8_t flags)
+{
+	if (err) {
+		LOG_DBG("VCP flags cb err (%d)", err);
+	}
+	LOG_DBG("VCP flags cb");
+}
+
+static struct bt_vcp_vol_rend_cb vcs_cb = {
+	.state = vcs_state_cb,
+	.flags = vcs_flags_cb,
+};
+
+void tester_handle_vcs(uint8_t opcode, uint8_t index, uint8_t *data,
+		       uint16_t len)
+{
+	switch (opcode) {
+	case VCS_READ_SUPPORTED_COMMANDS:
+		vcs_supported_commands(data, len);
+		break;
+	case VCS_SET_VOL:
+		set_volume(data);
+		break;
+	case VCS_VOL_UP:
+		vol_up();
+		break;
+	case VCS_VOL_DOWN:
+		vol_down();
+		break;
+	case VCS_MUTE:
+		mute();
+		break;
+	case VCS_UNMUTE:
+		unmute();
+		break;
+	default:
+		tester_rsp(BTP_SERVICE_ID_VCS, opcode, index,
+			   BTP_STATUS_UNKNOWN_CMD);
+		break;
+	}
+}
+
+/* Audio Input Control Service */
+static void aics_supported_commands(uint8_t *data, uint16_t len)
+{
+	struct net_buf_simple *buf = NET_BUF_SIMPLE(BTP_DATA_MAX_SIZE);
+
+	net_buf_simple_init(buf, 0);
+	net_buf_simple_add_u8(buf, AICS_READ_SUPPORTED_COMMANDS);
+	net_buf_simple_add_u8(buf, AICS_SET_GAIN);
+	net_buf_simple_add_u8(buf, AICS_MUTE);
+	net_buf_simple_add_u8(buf, AICS_UNMUTE);
+	net_buf_simple_add_u8(buf, AICS_MAN_GAIN);
+	net_buf_simple_add_u8(buf, AICS_AUTO_GAIN);
+	net_buf_simple_add_u8(buf, AICS_DESCRIPTION);
+
+	tester_send(BTP_SERVICE_ID_AICS, AICS_READ_SUPPORTED_COMMANDS,
+		    CONTROLLER_INDEX, buf->data, buf->len);
+}
+
+static void aics_state_cb(struct bt_aics *inst, int err, int8_t gain,
+			  uint8_t mute, uint8_t mode)
+{
+	LOG_DBG("AICS state callback");
+}
+
+static void aics_gain_setting_cb(struct bt_aics *inst, int err, uint8_t units,
+				 int8_t minimum, int8_t maximum)
+{
+	LOG_DBG("AICS gain setting callback");
+}
+
+static void aics_input_type_cb(struct bt_aics *inst, int err,
+			       uint8_t input_type)
+{
+	LOG_DBG("AICS input type callback");
+}
+
+static void aics_status_cb(struct bt_aics *inst, int err, bool active)
+{
+	LOG_DBG("AICS status callback");
+}
+
+static void aics_description_cb(struct bt_aics *inst, int err,
+				char *description)
+{
+	LOG_DBG("AICS description callback");
+}
```

Comment:
In the VCS callbacks, you also log the `err`. Should we do that here too?

Reply:
We are now.